### PR TITLE
upgrade react to latest version

### PR DESCRIPTION
**Pull request by fallynlogan**
upgrade react to latest version with command: yarn upgrade react@17.0.2 

need to upgrade react in order to upgrade microlink to resolve this error: 
 
Screen Shot 2021-06-01 at 1 09 37 PM

### CODE REVIEW

**Comment by cbarber** [comment]
:thinking: I'm a little surprised nothing else was upgraded.

**Comment by cbarber** [comment]
We should probably resolve the peer-dependency warnings as part of this upgrade:
```
warning " > @microlink/react@4.0.1" has incorrect peer dependency "react@>= 16 < 17".
warning "@microlink/react > nanoclamp@1.2.11" has incorrect peer dependency "react@>= 16 < 17".
...
warning " > react-apollo@2.5.6" has incorrect peer dependency "react@^15.0.0 || ^16.0.0".
warning " > react-dom@16.8.6" has incorrect peer dependency "react@^16.0.0".
warning " > react-highlight-words@0.16.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0-0".
warning " > styled-normalize@4.0.0" has incorrect peer dependency "react@>= 0.14.0 < 17.0.0-0".
```

I guess we don't need the other ticket since it'll happen all together.

**Comment by tmsavage** [comment]
Was there a specific change in the code that could have caused this issue where you needed to upgrade react?

**Comment by fallynlogan** [comment]
> Was there a specific change in the code that could have caused this issue where you needed to upgrade react?

@tmsavage I believe this error is caused by of the version of Microlink that we are using which is used to render the tweets. To upgrade Microlink and get rid of this error we have to upgrade react because the newer versions of Microlink require a newer version of react.

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "pluralize": "^8.0.0",
     "prop-types": "^15.6.1",
     "ramda": "^0.25.0",
-    "react": "^16.8.0",
+    "react": "17.0.2",
```
[the remark this answers is not in the document]
Comment (fallynlogan) [reply package.json:30]: What will that do? (or what does the ^ mean?)

[the remark this answers is not in the document]
Comment (chewnoill) [reply package.json:30]: https://bytearcher.com/articles/semver-explained-why-theres-a-caret-in-my-package-json/

File: package.json [hunk 2]
```diff
@@ -27,10 +27,12 @@
...
     "react-apollo": "^2.1.4",
-    "react-dom": "^16.8.6",
-    "react-highlight-words": "^0.16.0",
+    "react-dom": "17.0.2",
+    "react-highlight-words": "0.17.0",
+    "react-is": "17.0.2",
```
Comment (chewnoill) [inline package.json:34]: new unused dependencies are 😦

**Review by chewnoill — APPROVED** [review]

**Comment by cbarber** [comment]
Nice work. Some new warnings from apollo but that's fine for another ticket.

**Review by cbarber — APPROVED** [review]

**Pull request merged**